backend/app/processing/kb.py

```
@@ -0,0 +1,296 @@
+        metadata = {
+            "indexed_files": file_metadata,
+            "document_count": len(all_documents),
+            "text_chunks_count": len(all_document_texts),
+            "last_updated": str(datetime.datetime.now())
+        }
+        update_indexed_files_metadata(chatbot_id, metadata)
+        
+        logger.info(f"Built knowledge base for chatbot {chatbot_id} with {len(all_documents)} documents and {len(all_document_texts)} text chunks")
+        return True
+    
+    except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)